pymemcache/client/base.py

```diff
@@ -815,8 +818,6 @@ def _store_cmd(self, name, values, expire, noreply, cas=None):
             key = self.check_key(key)
             if self.serializer:
                 data, flags = self.serializer(key, data)
```

Comment:
It looks like this local `flags` variable will shadow the `flags` argument. How do they relate?

Reply:
I think this creates an unexpected case where `flags` is passed in but is overridden by the serializer. I can think of two improvements:

1. Because `flags` is a bit field, we could `OR` the two values together (the `flags` argument plus these `serializer_flags`) to form the final value send to the server. I don't know if that would always be the right thing to do, however, because we can't know in advance how the server and serializer are going to use the flags.
2. We could raise an exception when both `flags` and `self.serializer` are provided because they're incompatible.

Reply:
One more solution would be while using `data, flags = self.serializer(key, data)`, we could check if the `flags` is being passed, if `flags!=0`, we could use the flags generated by the `serializer`. Also, we could add more comments to tell the consumers about the logic of `serializer` and `flags` so they could be aware of the conflict passing `serializer` and `flags` at the same time.

Reply:
That sounds like a variant of (1) above (where we consider both input values and make a decision of which bits we want to send to the server). It has the same potential downsides, but given that we're baselining on `0` being "safe", it could be a good compromise.

Reply:
Which of these two params should have higher priority? In case they passed serializer and flags together, flags is not 0 generated from serialized, should we just ignore the flags passed by set function? This does not sound right to me.

Reply:
If we're going to choose one or the other, I think we should prefer the explicit parameter passed to the function.

But this is also why I was thinking we should consider merging the two bit fields (`flags | serialized_flags`) and sending that value to the server.

Reply:
for the bit operation, my concern would be, system might work fine if we just pass flags or serialized_flags, but if we do the bitwise operation, it could end up with exception. I would vote for using the flags passed by set cause that one should be the one we should use.

Reply:
I think that's fine. If someone is setting that value, they hopefully know what they're doing based on how their environment (memcached and chosen serializer) is configured.

Reply:
It does make me think that we might want to default `flags=None` in the signature, however, because it's possible someone might want to "force" `flags=0` to override the serializer's behavior. What do you think?

Reply:
the default behavior is `flags=0`. what you mentioned is a really corner case. I highly doubt this case would happen. but setting `flags=None` does make sense.  I will modify the code.